src/docs/arch-checkpoint.dox

```diff
@@ -19,46 +21,71 @@ WT_CONNECTION::rollback_to_stable.
 
 # The checkpoint algorithm #
 
-A checkpoint can be broken up into 5 main stages:
+The checkpoint algorithm can be considered as a coordinator of multiple aspects of the system. It
+relies on the @ref arch-btree "btree subsystem" for traversing in memory pages, the @ref arch-block "block manager subsystem" for
```

Comment:
I think "in memory" should be "in-memory". Please check for all uses of that.

Reply:
Also on this line I don't think we should be putting subsystems in double quotes like "btree subsystem". Just remove quotes. Many other arch pages talk about various subsystems without any quotes. There are a large number of them in these changes. Please change them all, I won't call each one out.

Reply:
The double quotes are for doxygen formatting fo the link, they don't actually appear in the compiled version.
In
> @ref arch-btree "btree subsystem"
- `arch-btree` is the tag to link to
- `btree subsystem` is the text to be displayed

Reply:
Great! Ignore my comments about that then...